lib/modules/apostrophe-ui/public/js/ui.js

```diff
@@ -264,6 +264,64 @@ apos.define('apostrophe-ui', {
       }, options));
     };
 
+    // Enhance a plaintext color field with Spectrum
+    self.enhanceColorpicker = function($el, options) {
+      var $preview = $el.prev();
+      var defaults = {
+        preview: true,
+        showButtons: true,
+        preferredFormat: "hex",
+        showInput: true,
+        color: $el.val(),
+        allowEmpty: true,
+        move: function(color) {
+
+          if (color) {
+            populateColorpicker();
+            $el.val(color.toHexString());
+          } else {
+            emptyColorpicker();
+          }
+
+          $el.trigger('change');
+        },
+        change: function(color) {
+          if (color) {
+            populateColorpicker();
+            $el.val(color.toHexString());
+          } else {
+            emptyColorpicker();
+          }
+        }
+      };
+  
+      var options = _.merge(defaults, options || {})
+
+      // initialize
+      $el.spectrum(options);
+      var $label = $el.next();
+
+      // set up preview on load
+      if ($el.spectrum('get')) {
+        populateColorpicker();
+      } else {
+        emptyColorpicker();
+      }
+
+      function emptyColorpicker() {
+        $label.text('None Selected');
```

Comment:
Nom this from a data attribute of the label or field that is set in the template, so it can go through localization with __()